tests/datasets/test_views.py

```diff
@@ -4251,3 +4257,64 @@ def test_dataset_rdf_download__datas_service(app: DjangoTestApp):
     </dcat:DataService>
 </rdf:RDF>"""
     )
+
+
+class TestDatasetDeleteView:
+    def test_delete_dataset(self, app: DjangoTestApp) -> None:
+        user = UserFactory(is_staff=True)
+        app.set_user(user)
+
+        dataset = DatasetFactory()
+
+        app.post(reverse("dataset-delete", args=[dataset.pk]))
+        assert not Dataset.objects.exists()
+
+
+class TestDeleteMemberView:
+    def test_remove_dataset_publisher_of_related_dataset_if_representative_is_deleted(
+        self, app: DjangoTestApp
+    ) -> None:
+        user = UserFactory(is_staff=True)
+        app.set_user(user)
+
+        organization = OrganizationFactory()
+        dataset = DatasetFactory(publisher=organization)
+        ct = ContentType.objects.get_for_model(dataset)
+        representative = RepresentativeFactory(
+            content_type=ct,
+            object_id=dataset.pk,
+            role=Representative.MANAGER,
+            organization=organization,
+        )
+
+        app.post(reverse("dataset-representative-delete", args=[dataset.pk, representative.pk]))
+
+        assert not Representative.objects.exists()
+        dataset.refresh_from_db()
+        assert dataset.publisher is None
+
+
+class TestRemoveRequestView:
+    def test_delete_dataset_comments_and_related_request_object(
+        self, app: DjangoTestApp
+    ) -> None:
+        user = UserFactory(is_staff=True)
+        app.set_user(user)
+
+        dataset = DatasetFactory()
+        request = RequestFactory(dataset=dataset)
+        CommentFactory(
+            rel_content_type=ContentType.objects.get_for_model(request),
+            rel_object_id=request.pk,
+        )
+        request_object = RequestObjectFactory(
+            request=request,
+            content_type=ContentType.objects.get_for_model(dataset),
+            object_id=dataset.pk,
+        )
+
+        app.post(reverse("dataset-request-remove", args=[dataset.pk, request_object.pk]))
+
+        assert Dataset.objects.exists()
```

Comment:
assert by id here as well
```suggestion
        assert Dataset.objects.filter(pk=dataset.pk).exists()
```

Reply:
Fixed in 797ada5082a2f6ca2ed2730bd11268a8a28e03cc